src/main/java/io/pravega/perf/PerfStats.java

```diff
@@ -202,12 +220,133 @@ private long windowTimeMS(long time) {
         }
     }
 
+    /***
+     * Perf Recorder that records the number of bytes every second.  Percentiles are reported in human form, i.e. MB/s
+     */
+    @NotThreadSafe
+    static private class ThroughtputWriter {
+        final String action;
+        final int messageSize;
+        final long startTime;
+        final List<Long> throughputs;
+        long count;
+        @GuardedBy("LOCk")
+        long seconds;
+        @GuardedBy("LOCk")
+        long totalBytesInSecond;
+        long totalBytes;
+
+        final ScheduledExecutorService scheduler;
+
+        private final Object LOCk = new Object();
+
+        ThroughtputWriter(String action, int messageSize, long startTime) {
+            this.action = action;
+            this.messageSize = messageSize;
+            this.startTime = startTime;
+            this.throughputs = new ArrayList<>();
+            this.count = 0;
+
+            scheduler = Executors.newSingleThreadScheduledExecutor();
+        }
+
+        public void start() {
+            scheduler.scheduleAtFixedRate(this::recordSecondThroughput, 1, 1, TimeUnit.SECONDS);
```

Comment:
You are assuming that this method actually runs exactly every second. I don't think this guarantees execution at a precise enough time. This will result in throughput rates that appear to oscillate when they are actually steady.
Instead, for each period you should calculate the number of ms actually elapsed and use that for the displayed throughput.